qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDbConnector.groovy

```diff
@@ -43,24 +43,34 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
    */
   @Override
   List<Customer> findCustomer(SearchCriteria criteria) {
-    String searchCondition = criteria.criteriaValue
 
-    switch(criteria.criteriaType){
-      case CriteriaType.LAST_NAME:
-        //todo create a Customer
-        List<Customer> customer = databaseQueries.findPersonByName(searchCondition)
-            return null
-      case CriteriaType.GROUP_NAME:
-            return databaseQueries.findCustomerByGroup(searchCondition)
-      case CriteriaType.ADD_ADDRESS:
-            return databaseQueries.findCustomerByAdditionalAddress(searchCondition)
-      case CriteriaType.CITY:
-            return databaseQueries.findCustomerByCity(searchCondition)
-      default:
-        //todo throw an exception
-        return null
+    List<Customer> foundCustomers = null
+
+    criteria.criteria.each { criteriaKey, criteriaValue ->
+
+      switch(criteriaKey){
+        case CriteriaType.FIRST_NAME:
+          foundCustomers.addAll(databaseQueries.findPersonByName(criteriaValue))
+          break
+        case CriteriaType.LAST_NAME:
+          foundCustomers.addAll(databaseQueries.findPersonByName(criteriaValue))
+          break
+        case CriteriaType.GROUP_NAME:
+          foundCustomers.addAll(databaseQueries.findCustomerByGroup(criteriaValue))
+          break
+        case CriteriaType.ADD_ADDRESS:
+          foundCustomers.addAll(databaseQueries.findCustomerByAdditionalAddress(criteriaValue))
+          break
+        case CriteriaType.CITY:
+          foundCustomers.addAll(databaseQueries.findCustomerByCity(criteriaValue))
+          break
+        default:
+          break
+      }
```

Comment:
Why not query the database for the needed information? DB queries make sense here since you can construct it in a way the filtering and merging are already done.

Reply:
I would leave it as is since we have a class with just the queries there. This is how it was before. I just added the first name attribute. I also think that this might not be part of my card since there is also a card for the database connector, which might be responsible for implementing this.